packages/payload/src/database/queryValidation/validateQueryPaths.ts

```diff
@@ -41,26 +42,66 @@ const flattenWhere = (query: Where): WhereField[] =>
     return [...flattenedConstraints, { [key]: val }]
   }, [])
 
+/**
+ * Iterates over the `where` object and to validate the field paths are correct and that the user has access to the fields
+ */
 export async function validateQueryPaths({
   collectionConfig,
   errors = [],
   globalConfig,
+  joins,
   overrideAccess,
   policies = {
     collections: {},
     globals: {},
   },
   req,
   versionFields,
-  where,
+  where: whereArg,
 }: Args): Promise<void> {
+  const where = whereArg
   const fields = flattenFields(
     versionFields || (globalConfig || collectionConfig).fields,
   ) as FieldAffectingData[]
+  const promises = []
+
+  // Validate the user has access to configured join fields
+  if (collectionConfig.joins) {
+    Object.entries(collectionConfig.joins).forEach(([collectionSlug, collectionJoins]) => {
+      collectionJoins.forEach((join) => {
+        if (join.field.where) {
+          promises.push(
+            validateQueryPaths({
```

Comment:
@r1tsuu I did this instead of merging the configured joins into the `JoinQuery` joins because we call this function multiple times. This could possibly be refactored or improved, but my head is spinning already.